content/en/docs/tutorials/services/pods-and-endpoint-termination-flow.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Pods And Endpoints Termination Flow
+content_type: tutorial
+weight: 60
+---
+
+
+<!-- overview -->
+
+Once you connected your Application with Service following steps
+like outlined in [Connecting Applications with Services](/docs/tutorials/services/connect-applications-service/),
+you have a continuously running, replicated application, that is exposed on a network.
+Let's look at Pods termination flow and ways to implement a graceful connections draining.
+
+# Pods And Endpoints Termination Flow
+
+There are often cases when you need to terminate a Pod - be it for upgrade or scale down.
+In order to improve application availability, it may be important to implement
+a proper active connections draining. This tutorial explains the flow of
+Pod termination in connection with the corresponding endpoint state and removal.
+
+This tutorial explains the flow of Pod termination in connection with the
+corresponding endpoint state and removal by using
+a simple nginx web server to demonstrate the concept.
+
+<!-- body -->
+
+## Example flow with endpoint termination
+
+The following is the example of the flow described in the
+[Termination of Pods](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)
+document.
+
+Let's say you have a Deployment containing of a single `nginx` replica
+(just for demonstration purposes) and a Service:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      terminationGracePeriodSeconds: 120
+      containers:
+      - name: nginx
+        image: nginx:latest
+        ports:
+        - containerPort: 80
+        lifecycle:
+          preStop:
+            exec:
+              # just hang around for the duration of terminationGracePeriodSeconds
+              command: [
+                "/bin/sh", "-c", "sleep 120"
```

Comment:
```suggestion
              # just hang around for at least the duration of terminationGracePeriodSeconds
              command: [
                "/bin/sh", "-c", "sleep 180"
```

to avoid people inferring that the two numbers somehow need to match

---

or, if they _do_ need to match: explain why.